### DISCUSSION TITLE
Add loading spinner to customer selection UI on order edit page

### DISCUSSION THREAD

**Discussion by jmauzyk** [craftcms/commerce#3826] [Ideas]
I’d like to suggest adding a loading spinner to the customer field UI on the order edit page. Our search performance has slowed over time due to the growing size of our search index DB table. As a result, it can take a few seconds for a potential customers to populate after entering an email in the customer field.

This delay has led to occasional confusion for our customer service agents. If they forget to wait for the search to finish and move the focus out of the customer field before selecting an existing customer or creating a new one, the customer selection UI won’t appear, making it difficult for them to update the order.

I think that adding a loading spinner to the customer selection UI could help mitigate this by indicating that the system is processing the search request. It would give agents visual feedback, letting them know the system is working and prompting them to wait for the results before proceeding with their selection.

While I’m actively working on optimizing our search index to improve its performance and manage its size, I believe this UI improvement has value regardless of query speed, as it provides clear feedback to users and helps prevent potential errors or confusion.

I’ve included a video demonstrating the delay in action for context.

I think this small improvement would reduce confusion and help streamline the workflow for customer service agents, particularly as our data grows.

Thanks for considering!

https://github.com/user-attachments/assets/09493502-fdef-4080-982c-fe625f5003b6

**Comment by lukeholder** [comment]
That's a good idea. Added to our list. Will update here once we have news to share. Thanks
